CMakeLists.txt

```diff
@@ -98,6 +98,22 @@ message("-- Python user site: ${PYTHON_USER_SITE}")
 message("-- pybind11 version: ${pybind11_VERSION}")
 message("-- pybind11 config: ${pybind11_CONFIG}")
 
+# When using python3 from XCode on MacOS, the RPath is wrongly set to XCode frameworks directory:
+#   LC_LOAD_DYLIB   @rpath/Python3.framework/Versions/3.7/Python3
+#   LC_RPATH        /Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.7/lib
+# Hence LC_RPATH/LC_LOAD_DYLIB does not exists.
+# Until this is fixed (not sure if it comes from pybind11, cmake or XCode), we can add another path to RPATH:
+#   LC_LOAD_DYLIB   @rpath/Python3.framework/Versions/3.7/Python3
+#   LC_RPATH        /Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.7/lib
+#                   /Applications/Xcode.app/Contents/Developer/Library/Frameworks  <----- ADDED
+# And now one combination of  LC_RPATH/LC_LOAD_DYLIB will be valid.
+# This should't change anything for those that use other python libs than XCode (homebrew for example) since the
+# LC_LOAD_DYLIB from XCode is quite unique.
+if (APPLE)
```

Comment:
I guess this is just a detail but if I want to use an homebrew version of python3 (lets say for the latest version for example),  the fact to force the rpath path to xcode wont mess everything  ? (as the test is "if APPLE")

Reply:
Hey @fredroy ,

It should not change anything for homebrew versions, since we can have multiple RPATH and brew installed version will be searching for `lib/libpython3.7.so` whereas XCode version is looking for `@rpath/Python3.framework/Versions/3.7/Python3`.

I will try to add both versions (XCode and homebrew) in the MacOS CI (next step after I finish ubuntu CI)